cmake/std/atdm/ats1/environment.sh

```diff
@@ -0,0 +1,200 @@
+################################################################################
+#
+# Set up env on ats1 for ATDM builds of Trilinos
+#
+# This source script gets the settings from the ATDM_CONFIG_BUILD_NAME var.
+#
+################################################################################
+
+if [ "$ATDM_CONFIG_COMPILER" == "DEFAULT" ] ; then
+  export ATDM_CONFIG_COMPILER=INTEL-19.0.4_MPICH-7.7.6
+fi
+
+if [ "$ATDM_CONFIG_KOKKOS_ARCH" == "DEFAULT" ] ; then
+  export ATDM_CONFIG_KOKKOS_ARCH=HSW
+fi
+
+export ATDM_CONFIG_ENABLE_SPARC_SETTINGS=ON
+export ATDM_CONFIG_USE_NINJA=ON
+
+# Use srun to run mpi jobs
+export ATDM_CONFIG_MPI_EXEC="/opt/slurm/bin/srun"
+
+# srun does not accept "-np" for # of processors
+export ATDM_CONFIG_MPI_EXEC_NUMPROCS_FLAG="--ntasks"
+# Use pmi2 as MPI type.
+# Ensure that no more than 36 tasks, per srun command, are launched.
+export ATDM_CONFIG_MPI_PRE_FLAGS="--mpi=pmi2;--ntasks-per-node;36"
+
+# Common sparc tpl path values
+sparc_tpl_prefix_path="/usr/projects/sparc/tpls"
+system_name="ats1"
+node_arch=""
+
+if [[ "$ATDM_CONFIG_KOKKOS_ARCH" == "HSW" ]]; then
+  node_arch="hsw"
+  module unload craype-mic-knl
+  module load craype-haswell
+  # HSW nodes have 64 virtual cores.
+  # Allow no more than 8 virtual cores per task.
+  export ATDM_CONFIG_MPI_POST_FLAGS="-c 4"
+  # If we have 1 MPI rank per srun command and 1 cpu per task, we can run up to 32 2-threaded tests "in parallel" on virtual cores.
+  export ATDM_CONFIG_CTEST_PARALLEL_LEVEL=1
+  # Assume we are building on a compute node
+  export ATDM_CONFIG_BUILD_COUNT=64
+elif [[ "$ATDM_CONFIG_KOKKOS_ARCH" == "KNL" ]]; then
+  node_arch="knl"
+  module unload craype-haswell
+  module load craype-mic-knl
+  export SLURM_TASKS_PER_NODE=16
+  export OMP_PLACES=threads
+  export OMP_PROC_BIND=spread
+  # Ensure that no more than 8 tasks, per srun command, are launched.
+  export ATDM_CONFIG_MPI_PRE_FLAGS="--mpi=pmi2;--ntasks-per-node=8"
+  # KNL nodes have 272 virtual cores.
+  # Allow now more than 34 virtual cores per task.
+  export ATDM_CONFIG_MPI_POST_FLAGS="--hint=nomultithread;-c 4"
+  export ATDM_CONFIG_CTEST_PARALLEL_LEVEL=1
+  # Assume we are building on a compute node
+  export ATDM_CONFIG_BUILD_COUNT=272
```

Comment:
The compute nodes on 'mutrino' can really handle 272 parallel object-file builds and links?